session_test.go

```diff
@@ -135,6 +135,25 @@ func (s *S) TestURLParsing(c *C) {
 	}
 }
 
+func (s *S) TestURLSsl(c *C) {
+	type test struct {
+		url           string
+		nilDialServer bool
+	}
+
+	tests := []test{
+		{"localhost:40001", true},
+		{"localhost:40001?ssl=false", true},
+		{"localhost:40001?ssl=true", false},
+	}
+
+	for _, test := range tests {
+		info, err := mgo.ParseURL(test.url)
```

Comment:
It would be nice to dial to the SSL-enabled node in the test harness (`localhost:40003`) and perform a `Ping()` to confirm the SSL is working 👍

Reply:
Can you give me an example ? I'm not used to test harness.

Also how can I run the tests locally please ?

Reply:
Sure! Something like:

```go
session, err := mgo.Dial(test.url)
c.Assert(err, IsNil)
c.Assert(session.Ping(), IsNil)
```

Running the tests locally can be tricky if you're on a BSD derivative like OS X, but if you're on linux you should be OK with installing `supervisord` and running `make startdb` in the mgo root.

Reply:
done aa05fa4

(wasn't able to run it locally, even with `supervisord` installed, I must have miss something)